docs/upgrade.html

```diff
@@ -21,6 +21,10 @@
 0.10.0.0 has <a href="#upgrade_10_breaking">potential breaking changes</a> (please review before upgrading) and
 there may be a <a href="#upgrade_10_performance_impact">performance impact during the upgrade</a>. Because new protocols
 are introduced, it is important to upgrade your Kafka clusters before upgrading your clients.
+<p/>
+<b>Notes to client with version 0.9.0.0: </b>Due to a bug introduced in 0.9.0.0,
+clients with ZooKeeper dependency may not be able to work with 0.10.0.0 broker. Therefore, 0.9.0.0 clients should
```

Comment:
I don't agree that we achieve the same goal. From a storage format documentation perspective, it's easier to track format changes if there is a version associated with them (even when doing compatible changes like adding fields). It is still useful to know when a particular set of fields was added ((in this case it's just a single field, but in others there will be more).

Yet another way to handle that is to tie the format with `ApiVersion`. This can be done implicitly or explicitly.